causync.py

```diff
@@ -97,18 +127,18 @@ def run_sync(self):
         dst = os.path.abspath(os.path.join(self.dst_abs, self.curdate))
         cmd = "rsync {f} {ef} {src} {dst}".format(f=self.config.RSYNC_FLAGS,
                                                   ef=extra_flags,
-                                                  src=self.src_abs,
+                                                  src=" ".join(self.src_abs),
                                                   dst=dst)
 
         self.logger.debug("rsync cmd = {}".format(cmd))
 
         self.logger.info("syncing {} to {}".format(self.src_abs, dst))
-        self.create_lockfile()
+        self.create_lockfiles()
 
         result = subprocess.check_output(cmd, shell=True).decode()
         self.logger.debug(result)
 
-        self.remove_lockfile()
+        self.remove_lockfiles()
```

Comment:
This should be run in a try...finally block, so you remove lock files on a keyboard interrupt or on receiving a SIGTERM signal.

Reply:
Good idea, I'll add a SIGTERM/SIGINT handler.